numba/targets/arrayobj.py

```diff
@@ -1291,25 +1295,308 @@ def _empty_nd_impl(context, builder, arrtype, shapes):
                    strides=cgutils.pack_array(builder, strides),
                    itemsize=itemsize,
                    meminfo=meminfo)
-    return ary._getvalue()
+    return ary
 
-
-@builtin
-@implement(numpy.empty, types.Kind(types.Integer))
-@implement(numpy.empty, types.Kind(types.BaseTuple))
-@implement(numpy.empty, types.Kind(types.Integer), types.Kind(types.Function))
-@implement(numpy.empty, types.Kind(types.BaseTuple), types.Kind(types.Function))
-def numpy_empty_nd(context, builder, sig, args):
+def _zero_fill_array(context, builder, ary):
+    """
+    Zero-fill an array.  The array must be contiguous.
+    """
+    memset = "llvm.memset.p0i8.i%d" % (types.intp.bitwidth)
```

Comment:
Is it worth making a `memset` function in cgutils? It's somewhere that I'd check for it if I wanted to use `memset` from somewhere else, but might miss it in here.